commons-math-legacy/src/test/java/org/apache/commons/math4/legacy/analysis/interpolation/BicubicInterpolatingFunctionTest.java

```diff
@@ -304,7 +304,7 @@ public void testSplinePartialDerivatives() {
             }
         }
 
-        final BicubicFunction f = new BicubicFunction(coeff, true);
+        final BicubicFunction f = new BicubicFunction(coeff, 1, 1, true);
```

Comment:
Have you tested this when `xR` and `yR` are not 1? I would suggest simply repeating the test with `xR` and `yR` as  0.5 and 2 then testing the derivative is correctly scaled.

Reply:
That uses an interval of 1 / 20 for both axes. Perhaps update to have different step sizes for each axis. This will ensure xR and yR are not mixed up.

Reply:
Done ! I updated that test to have different xval and yval array, and it's still working.